### PR TITLE
NIFI-10306: Add logging to StandardProcessGroup.addConnection() with more details

### PR DESCRIPTION
**Pull request by mattyb149**
# Summary

[NIFI-10306](https://issues.apache.org/jira/browse/NIFI-10306) This PR adds logging to better identify the components of a Connection that cannot be added to the flow, to include source and destination process groups and types, plus the reason.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [x] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java [hunk 1]
```diff
@@ -1146,42 +1146,75 @@ public void addConnection(final Connection connection) {
...
+                                "] because its destination [" + destination.getIdentifier() +
+                                "] is an Input Port but the Input Port does not belong to a child Process Group");
                     }
                 } else if (destinationGroup != this) {
-                    throw new IllegalStateException("Cannot add Connection between " + source.getIdentifier() + " and " + destination.getIdentifier()
-                        + " because they are in different Process Groups and neither is an Input Port or Output Port");
+                    throw new IllegalStateException("Cannot add Connection from " + source.getConnectableType().name() + "[" + source.getIdentifier() +
+                            "] from Process Group [" + sourceGroup.getIdentifier() +
+                            "] to Process Group [" + destinationGroup.getIdentifier() +
+                            "] destination " + destination.getConnectableType().name() + "[" + destination.getIdentifier() +
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java:1216]: Is there a reason for including the Connectable Type name value in this message but not the others?

In light of the similarities, what do you think about creating a simple method for formatting the source and destination messages to ensure consistency?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the improvement @mattyb149, having these additional details in the exception messages should be helpful.

Just noted a question on one of the messages, and the opportunity to create a reusable method to ensure consistent message formatting.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java (hunk 1 above)
Comment (mattyb149) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java:1216]: The others should have the connectable type hard-coded into the string. I can change them all in order to externalize a method that takes the groups and such as arguments and return the formatted string.

Comment (mattyb149) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java:1216]: Many of the messages are slightly different, I think the extracted method would be more awkward as you'd have to include multiple strings to be inserted in various spots. I'd prefer to keep this as-is if that's ok

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java:1267]: Thanks for evaluating the options @mattyb149, on further review, keeping the messages as implemented sounds good.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the improvements @mattyb149! +1 merging

**Pull request closed without merging**